pkg/mssqlcontainer/ingest/extract/7zip.go

```diff
@@ -0,0 +1,95 @@
+package extract
+
+import (
+	"github.com/microsoft/go-sqlcmd/internal/container"
+	"path/filepath"
+	"regexp"
+)
+
+type sevenZip struct {
+	controller  *container.Controller
+	containerId string
+}
+
+func (e *sevenZip) Initialize(controller *container.Controller) {
```

Comment:
> Initialize(controller *container.Controller) [](http://example.com/codeflow?start=18&length=45)

How can we abstract away the differences between
"SQL running in a container", "SQL running on the local host",  "SQL running in the cloud", and "SQL running on a remote host"? 
Or would we have multiple extractor implementations and this one should be renamed "sevenZipInLinuxX64Container" and we'd have others like "windowsZipLocalHost" ?

I don't see how this model evolves to support the non-container scenarios.